### PR TITLE
CRB-144 Port changes from C++ SDK to Rust SDK, add minor fixes

### PR DESCRIPTION
**Pull request by nathanwhit**

### CODE REVIEW

File: protos/processor.proto
Comment (CAGS295) [inline protos/processor.proto]: nitpick, keep comments style consistent.

**Comment by CAGS295** [comment]
[CRB-144]

[CRB-144]: https://gluwa.atlassian.net/browse/CRB-144

**Review by CAGS295 — APPROVED** [review]

**Comment by CAGS295** [comment]
target `dev`

**Comment by CAGS295** [comment]
@atodorov do you want to add CI before I merge this PR?

**Comment by atodorov** [comment]
> @atodorov do you want to add CI before I merge this PR?

Preferably yes, see #2. 

I've copied the test setup which they have in Jenkinsfile which uses containers. If that doesn't work then I'll look into something else, at least adding a compile & lint jobs for some sanity testing.

**Review by atodorov — DISMISSED** [review]
Please rebase so we can see the CI status.

**Comment by nathanwhit** [comment]
Hm looks like CI failed with this line:
```
intkey-tp-rust_1       | Please build intkey-tp-rust first with 'docker-compose -f docker/compose/sawtooth-build.yaml up'
```
Not quite sure what to make of it at first glance, since I don't see any compilation failures or issues directly related to my changes. I'll look into it. cc @atodorov

**Comment by atodorov** [comment]
> Hm looks like CI failed with this line:
> 
> ```
> intkey-tp-rust_1       | Please build intkey-tp-rust first with 'docker-compose -f docker/compose/sawtooth-build.yaml up'
> ```
> 
It's not that. This means the doker image we need hasn't been built but b/c the docker-compose file contains instructions for that it gets built on the fly. The error that I see is: 

```
2021-09-27T16:45:36.7436495Z [33mxo-tp-rust_1      |[0m [0;32m--- Building xo-tp-rust ---
2021-09-27T16:45:36.7449354Z [33mxo-tp-rust_1      |[0m [0m
2021-09-27T16:45:37.1687442Z [33mxo-tp-rust_1      |[0m     Updating crates.io index
2021-09-27T16:45:37.1774728Z [36mintkey-tp-rust_1  |[0m     Updating crates.io index
2021-09-27T16:45:54.8480836Z [33mxo-tp-rust_1      |[0m error: no matching package named `sawtooth-sdk` found
2021-09-27T16:45:54.8482033Z [33mxo-tp-rust_1      |[0m location searched: /project/sawtooth-sdk-rust
2021-09-27T16:45:54.8483397Z [33mxo-tp-rust_1      |[0m required by package `sawtooth-xo-tp-rust v0.5.2 (/project/sawtooth-sdk-rust/examples/xo_rust)`
2021-09-27T16:45:55.3990206Z [33m1279151612_xo-tp-rust_1 exited with code 101
2021-09-27T16:45:55.5078282Z [0m[36mintkey-tp-rust_1  |[0m error: no matching package named `sawtooth-sdk` found
```

Which could be a version mismatch from upstream or maybe using a newer version of the compiler.  

Definitely not related to your changes but ideally we'd like to see CI jobs passing before merging.

**Comment by nathanwhit** [comment]
Whoops you're right, I was looking in the failed step, but the actual failure was in the step prior. I actually do know the cause of that (it's because the package is renamed so we avoid accidentally pulling the non forked SDK, but the examples still referred to the old package name). Pushed a fix.

**Comment by nathanwhit** [comment]
CI is green, so this just needs another review before merging.

**Review by seandeneen-gluwa — APPROVED** [review]

**Pull request merged**
